In [7]:
# NVIDIA-SMI
! nvidia-smi

Thu Dec 26 10:13:44 2024       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 525.105.17   Driver Version: 525.105.17   CUDA Version: 12.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  Off  | 00000000:01:00.0 Off |                  N/A |
| 67%   71C    P2   268W / 350W |   1714MiB / 24576MiB |     97%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-------

In [10]:
# GPU information
! nvidia-smi --query-gpu=gpu_name --format=csv

name
NVIDIA GeForce RTX 3090


In [11]:
# CPU information
! cat /proc/cpuinfo

processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 165
model name	: Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz
stepping	: 5
microcode	: 0xfc
cpu MHz		: 4800.546
cache size	: 20480 KB
physical id	: 0
siblings	: 20
core id		: 0
cpu cores	: 10
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 22
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsav

In [21]:
# OS information
! cat /etc/issue  

Ubuntu 20.04.5 LTS \n \l



In [43]:
import torch 
print(torch.__version__)

2.1.0


In [48]:
import pytoda
print(pytoda.__version__)

1.1.3


In [ ]:
# Install the required packages
!pip install -r requirements.txt

In [18]:
!pip 

/bin/bash: pip: command not found


ubuntu@wuyang:~$ conda list
# packages in environment at /home/ubuntu/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             5.1                       1_gnu
boltons                   23.0.0          py311h06a4308_0
brotlipy                  0.7.0           py311h5eee18b_1002
bzip2                     1.0.8                h7b6447c_0
c-ares                    1.19.1               h5eee18b_0
ca-certificates           2023.08.22           h06a4308_0
certifi                   2023.7.22       py311h06a4308_0
cffi                      1.15.1          py311h5eee18b_3
charset-normalizer        2.0.4              pyhd3eb1b0_0
conda                     23.9.0          py311h06a4308_0
conda-content-trust       0.2.0           py311h06a4308_0
conda-libmamba-solver     23.9.1          py311h06a4308_0
conda-package-handling    2.2.0           py311h06a4308_0
conda-package-streaming   0.9.0           py311h06a4308_0
cryptography              41.0.3          py311hdda0065_0
fmt                       9.1.0                hdb19cb5_0
icu                       73.1                 h6a678d5_0
idna                      3.4             py311h06a4308_0
jsonpatch                 1.32               pyhd3eb1b0_0
jsonpointer               2.1                pyhd3eb1b0_0
krb5                      1.20.1               h143b758_1
ld_impl_linux-64          2.38                 h1181459_1
libarchive                3.6.2                h6ac8c49_2
libcurl                   8.2.1                h251f7ec_0
libedit                   3.1.20221030         h5eee18b_0
libev                     4.33                 h7f8727e_1
libffi                    3.4.4                h6a678d5_0
libgcc-ng                 11.2.0               h1234567_1
libgomp                   11.2.0               h1234567_1
libmamba                  1.5.1                haf1ee3a_0
libmambapy                1.5.1           py311h2dafd23_0
libnghttp2                1.52.0               h2d74bed_1
libsolv                   0.7.24               he621ea3_0
libssh2                   1.10.0               hdbd6064_2
libstdcxx-ng              11.2.0               h1234567_1
libuuid                   1.41.5               h5eee18b_0
libxml2                   2.10.4               hf1b16e4_1
lz4-c                     1.9.4                h6a678d5_0
ncurses                   6.4                  h6a678d5_0
openssl                   3.0.11               h7f8727e_2
packaging                 23.1            py311h06a4308_0
pcre2                     10.42                hebb0a14_0
pip                       23.2.1          py311h06a4308_0
pluggy                    1.0.0           py311h06a4308_1
pybind11-abi              4                    hd3eb1b0_1
pycosat                   0.6.6           py311h5eee18b_0
pycparser                 2.21               pyhd3eb1b0_0
pyopenssl                 23.2.0          py311h06a4308_0
pysocks                   1.7.1           py311h06a4308_0
python                    3.11.5               h955ad1f_0
readline                  8.2                  h5eee18b_0
reproc                    14.2.4               h295c915_1
reproc-cpp                14.2.4               h295c915_1
requests                  2.31.0          py311h06a4308_0
ruamel.yaml               0.17.21         py311h5eee18b_0
setuptools                68.0.0          py311h06a4308_0
sqlite                    3.41.2               h5eee18b_0
tk                        8.6.12               h1ccaba5_0
tqdm                      4.65.0          py311h92b7b1e_0
truststore                0.8.0           py311h06a4308_0
tzdata                    2023c                h04d1e81_0
urllib3                   1.26.16         py311h06a4308_0
wheel                     0.41.2          py311h06a4308_0
xz                        5.4.2                h5eee18b_0
yaml-cpp                  0.7.0                h295c915_1
zlib                      1.2.13               h5eee18b_0
zstandard                 0.19.0          py311h5eee18b_0
zstd                      1.5.5                hc292b87_0

# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             5.1                       1_gnu
alembic                   1.14.0                   pypi_0    pypi
anyio                     4.0.0                    pypi_0    pypi
argon2-cffi               23.1.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
arrow                     1.3.0                    pypi_0    pypi
asttokens                 2.4.1                    pypi_0    pypi
async-lru                 2.0.4                    pypi_0    pypi
attrs                     23.1.0                   pypi_0    pypi
autograd                  1.7.0                    pypi_0    pypi
autograd-gamma            0.5.0                    pypi_0    pypi
babel                     2.13.1                   pypi_0    pypi
beautifulsoup4            4.12.2                   pypi_0    pypi
blas                      1.0                         mkl
bleach                    6.1.0                    pypi_0    pypi
bottleneck                1.3.7           py310ha9d4c09_0
brotli                    1.0.9                h5eee18b_7
brotli-bin                1.0.9                h5eee18b_7
brotli-python             1.0.9           py310h6a678d5_7
bzip2                     1.0.8                h7b6447c_0
ca-certificates           2024.3.11            h06a4308_0
certifi                   2024.2.2        py310h06a4308_0
cffi                      1.15.1          py310h5eee18b_3
charset-normalizer        2.0.4              pyhd3eb1b0_0
colorlog                  6.9.0                    pypi_0    pypi
comm                      0.2.0                    pypi_0    pypi
contourpy                 1.2.0           py310hdb19cb5_0
cryptography              41.0.3          py310hdda0065_0
cuda-cudart               12.1.105                      0    nvidia
cuda-cupti                12.1.105                      0    nvidia
cuda-libraries            12.1.0                        0    nvidia
cuda-nvrtc                12.1.105                      0    nvidia
cuda-nvtx                 12.1.105                      0    nvidia
cuda-opencl               12.3.52                       0    nvidia
cuda-runtime              12.1.0                        0    nvidia
cycler                    0.11.0             pyhd3eb1b0_0
cyrus-sasl                2.1.28               h52b45da_1
dbus                      1.13.18              hb2f20db_0
debugpy                   1.8.0                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
dill                      0.3.7                    pypi_0    pypi
diskcache                 5.6.3                    pypi_0    pypi
exceptiongroup            1.1.3                    pypi_0    pypi
executing                 2.0.1                    pypi_0    pypi
expat                     2.5.0                h6a678d5_0
fastjsonschema            2.18.1                   pypi_0    pypi
fastprogress              1.0.3                    pypi_0    pypi
ffmpeg                    4.3                  hf484d3e_0    pytorch
filelock                  3.9.0           py310h06a4308_0
fontconfig                2.14.1               h4c34cd2_2
fonttools                 4.25.0             pyhd3eb1b0_0
formulaic                 1.0.2                    pypi_0    pypi
fqdn                      1.5.1                    pypi_0    pypi
freetype                  2.12.1               h4a9f257_0
fsspec                    2023.10.0                pypi_0    pypi
gensim                    4.3.2                    pypi_0    pypi
giflib                    5.2.1                h5eee18b_3
glib                      2.78.4               h6a678d5_0
glib-tools                2.78.4               h6a678d5_0
gmp                       6.2.1                h295c915_3
gmpy2                     2.1.2           py310heeb90bb_0
gnutls                    3.6.15               he1e5248_0
greenlet                  3.1.1                    pypi_0    pypi
gst-plugins-base          1.14.1               h6a678d5_1
gstreamer                 1.14.1               h5eee18b_1
icu                       73.1                 h6a678d5_0
idna                      3.4             py310h06a4308_0
importlib-metadata        6.8.0                    pypi_0    pypi
importlib-resources       6.1.1                    pypi_0    pypi
intel-openmp              2023.1.0         hdb19cb5_46305
interface-meta            1.3.0                    pypi_0    pypi
ipykernel                 6.26.0                   pypi_0    pypi
ipython                   8.17.2                   pypi_0    pypi
ipywidgets                8.1.1                    pypi_0    pypi
isoduration               20.11.0                  pypi_0    pypi
jedi                      0.19.1                   pypi_0    pypi
jinja2                    3.1.2           py310h06a4308_0
joblib                    1.3.2                    pypi_0    pypi
jpeg                      9e                   h5eee18b_1
json5                     0.9.14                   pypi_0    pypi
jsonpointer               2.4                      pypi_0    pypi
jsonschema                4.19.2                   pypi_0    pypi
jsonschema-specifications 2023.7.1                 pypi_0    pypi
jupyter                   1.0.0                    pypi_0    pypi
jupyter-client            8.6.0                    pypi_0    pypi
jupyter-console           6.6.3                    pypi_0    pypi
jupyter-core              5.5.0                    pypi_0    pypi
jupyter-events            0.9.0                    pypi_0    pypi
jupyter-lsp               2.2.0                    pypi_0    pypi
jupyter-server            2.10.0                   pypi_0    pypi
jupyter-server-terminals  0.4.4                    pypi_0    pypi
jupyterlab                4.0.8                    pypi_0    pypi
jupyterlab-pygments       0.2.2                    pypi_0    pypi
jupyterlab-server         2.25.0                   pypi_0    pypi
jupyterlab-widgets        3.0.9                    pypi_0    pypi
kiwisolver                1.4.4           py310h6a678d5_0
krb5                      1.20.1               h143b758_1
lame                      3.100                h7b6447c_0
lcms2                     2.12                 h3be6417_0
ld_impl_linux-64          2.38                 h1181459_1
lerc                      3.0                  h295c915_0
libbrotlicommon           1.0.9                h5eee18b_7
libbrotlidec              1.0.9                h5eee18b_7
libbrotlienc              1.0.9                h5eee18b_7
libclang                  14.0.6          default_hc6dbbc7_1
libclang13                14.0.6          default_he11475f_1
libcublas                 12.1.0.26                     0    nvidia
libcufft                  11.0.2.4                      0    nvidia
libcufile                 1.8.0.34                      0    nvidia
libcups                   2.4.2                h2d74bed_1
libcurand                 10.3.4.52                     0    nvidia
libcusolver               11.4.4.55                     0    nvidia
libcusparse               12.0.2.55                     0    nvidia
libdeflate                1.17                 h5eee18b_1
libedit                   3.1.20230828         h5eee18b_0
libffi                    3.4.4                h6a678d5_0
libgcc-ng                 11.2.0               h1234567_1
libglib                   2.78.4               hdc74915_0
libgomp                   11.2.0               h1234567_1
libiconv                  1.16                 h7f8727e_2
libidn2                   2.3.4                h5eee18b_0
libjpeg-turbo             2.0.0                h9bf148f_0    pytorch
libllvm14                 14.0.6               hdb19cb5_3
libnpp                    12.0.2.50                     0    nvidia
libnvjitlink              12.1.105                      0    nvidia
libnvjpeg                 12.1.1.14                     0    nvidia
libpng                    1.6.39               h5eee18b_0
libpq                     12.17                hdbd6064_0
libstdcxx-ng              11.2.0               h1234567_1
libtasn1                  4.19.0               h5eee18b_0
libtiff                   4.5.1                h6a678d5_0
libunistring              0.9.10               h27cfd23_0
libuuid                   1.41.5               h5eee18b_0
libwebp                   1.3.2                h11a3e52_0
libwebp-base              1.3.2                h5eee18b_0
libxcb                    1.15                 h7f8727e_0
libxkbcommon              1.0.1                h5eee18b_1
libxml2                   2.10.4               hf1b16e4_1
lifelines                 0.30.0                   pypi_0    pypi
lightgbm                  4.5.0                    pypi_0    pypi
llvm-openmp               14.0.6               h9e868ea_0
lz4-c                     1.9.4                h6a678d5_0
mako                      1.3.6                    pypi_0    pypi
markupsafe                2.1.1           py310h7f8727e_0
matplotlib                3.8.0           py310h06a4308_0
matplotlib-base           3.8.0           py310h1128e8f_0
matplotlib-inline         0.1.6                    pypi_0    pypi
mistune                   3.0.2                    pypi_0    pypi
mkl                       2023.1.0         h213fc3f_46343
mkl-service               2.4.0           py310h5eee18b_1
mkl_fft                   1.3.8           py310h5eee18b_0
mkl_random                1.2.4           py310hdb19cb5_0
mpc                       1.1.0                h10f8cd9_1
mpfr                      4.0.2                hb69a4c5_1
mpmath                    1.3.0           py310h06a4308_0
munkres                   1.1.4                      py_0
mysql                     5.7.24               h721c034_2
nbclient                  0.9.0                    pypi_0    pypi
nbconvert                 7.11.0                   pypi_0    pypi
nbformat                  5.9.2                    pypi_0    pypi
ncurses                   6.4                  h6a678d5_0
nest-asyncio              1.5.8                    pypi_0    pypi
nettle                    3.7.3                hbbd107a_1
networkx                  3.1             py310h06a4308_0
notebook                  7.0.6                    pypi_0    pypi
notebook-shim             0.2.3                    pypi_0    pypi
numexpr                   2.8.7           py310h85018f9_0
numpy                     1.26.0          py310h5f9d8c6_0
numpy-base                1.26.0          py310hb5e798b_0
nvidia-nccl-cu12          2.23.4                   pypi_0    pypi
openh264                  2.1.1                h4ff587b_0
openjpeg                  2.4.0                h3ad879b_0
openssl                   3.0.13               h7f8727e_0
optuna                    4.1.0                    pypi_0    pypi
overrides                 7.4.0                    pypi_0    pypi
packaging                 23.2            py310h06a4308_0
pandas                    2.1.2                    pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pcre2                     10.42                hebb0a14_0
pexpect                   4.8.0                    pypi_0    pypi
pillow                    10.0.1          py310ha6cbd5a_0
pip                       23.3            py310h06a4308_0
platformdirs              3.11.0                   pypi_0    pypi
ply                       3.11            py310h06a4308_0
prometheus-client         0.18.0                   pypi_0    pypi
prompt-toolkit            3.0.39                   pypi_0    pypi
psutil                    5.9.6                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pubchempy                 1.0.4                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
pycparser                 2.21               pyhd3eb1b0_0
pyfaidx                   0.7.2.2                  pypi_0    pypi
pygments                  2.16.1                   pypi_0    pypi
pyopenssl                 23.2.0          py310h06a4308_0
pyparsing                 3.0.9           py310h06a4308_0
pyqt                      5.15.10         py310h6a678d5_0
pyqt5-sip                 12.13.0         py310h5eee18b_0
pysocks                   1.7.1           py310h06a4308_0
python                    3.10.13              h955ad1f_0
python-dateutil           2.8.2              pyhd3eb1b0_0
python-json-logger        2.0.7                    pypi_0    pypi
python-tzdata             2023.3             pyhd3eb1b0_0
pytoda                    1.1.3                    pypi_0    pypi
pytorch                   2.1.0           py3.10_cuda12.1_cudnn8.9.2_0    pytorch
pytorch-cuda              12.1                 ha16c6d3_5    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2023.3.post1    py310h06a4308_0
pyyaml                    6.0.1           py310h5eee18b_0
pyzmq                     25.1.1                   pypi_0    pypi
qt-main                   5.15.2              h53bd1ea_10
qtconsole                 5.5.0                    pypi_0    pypi
qtpy                      2.4.1                    pypi_0    pypi
rdkit-pypi                2022.9.5                 pypi_0    pypi
readline                  8.2                  h5eee18b_0
referencing               0.30.2                   pypi_0    pypi
requests                  2.31.0          py310h06a4308_0
rfc3339-validator         0.1.4                    pypi_0    pypi
rfc3986-validator         0.1.1                    pypi_0    pypi
rpds-py                   0.12.0                   pypi_0    pypi
scikit-learn              1.3.2                    pypi_0    pypi
scipy                     1.11.3                   pypi_0    pypi
seaborn                   0.12.2          py310h06a4308_0
selfies                   2.1.1                    pypi_0    pypi
send2trash                1.8.2                    pypi_0    pypi
setuptools                68.0.0          py310h06a4308_0
sip                       6.7.12          py310h6a678d5_0
six                       1.16.0             pyhd3eb1b0_1
smart-open                6.4.0                    pypi_0    pypi
smilespe                  0.0.3                    pypi_0    pypi
sniffio                   1.3.0                    pypi_0    pypi
some-package              0.1                      pypi_0    pypi
soupsieve                 2.5                      pypi_0    pypi
sqlalchemy                2.0.36                   pypi_0    pypi
sqlite                    3.41.2               h5eee18b_0
stack-data                0.6.3                    pypi_0    pypi
sympy                     1.11.1          py310h06a4308_0
tbb                       2021.8.0             hdb19cb5_0
terminado                 0.17.1                   pypi_0    pypi
threadpoolctl             3.2.0                    pypi_0    pypi
tinycss2                  1.2.1                    pypi_0    pypi
tk                        8.6.12               h1ccaba5_0
tomli                     2.0.1           py310h06a4308_0
torchaudio                2.1.0               py310_cu121    pytorch
torchtriton               2.1.0                     py310    pytorch
torchvision               0.16.0              py310_cu121    pytorch
tornado                   6.3.3           py310h5eee18b_0
tqdm                      4.66.1                   pypi_0    pypi
traitlets                 5.13.0                   pypi_0    pypi
types-python-dateutil     2.8.19.14                pypi_0    pypi
typing_extensions         4.7.1           py310h06a4308_0
tzdata                    2023c                h04d1e81_0
unidecode                 1.3.7                    pypi_0    pypi
upfp                      0.0.5                    pypi_0    pypi
uri-template              1.3.0                    pypi_0    pypi
urllib3                   1.26.18         py310h06a4308_0
wcwidth                   0.2.9                    pypi_0    pypi
webcolors                 1.13                     pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket-client          1.6.4                    pypi_0    pypi
wheel                     0.41.2          py310h06a4308_0
widgetsnbextension        4.0.9                    pypi_0    pypi
wrapt                     1.17.0                   pypi_0    pypi
xgboost                   2.1.2                    pypi_0    pypi
xz                        5.4.2                h5eee18b_0
yaml                      0.2.5                h7b6447c_0
zipp                      3.17.0                   pypi_0    pypi
zlib                      1.2.13               h5eee18b_0
zstd                      1.5.5                hc292b87_0